src/SAML2/XML/samlp/ArtifactResolve.php

```diff
@@ -65,6 +82,54 @@ public function setArtifact(string $artifact): void
     }
 
 
+    /**
+     * Create a class from XML
+     *
+     * @param \DOMElement $xml
+     * @return self
+     */
+    public static function fromXML(DOMElement $xml): object
+    {
+        Assert::same($xml->localName, 'ArtifactResolve');
+        Assert::same($xml->namespaceURI, ArtifactResolve::NS);
+
+        $id = self::getAttribute($xml, 'ID');
+        $version = self::getAttribute($xml, 'Version');
+        $issueInstant = Utils::xsDateTimeToTimestamp(self::getAttribute($xml, 'IssueInstant'));
+        $destination = self::getAttribute($xml, 'Destination', null);
+        $consent = self::getAttribute($xml, 'Consent', null);
+
+        $issuer = Issuer::getChildrenOfClass($xml);
+        Assert::countBetween($issuer, 0, 1);
```

Comment:
I guess this could be `maxCount()` instead, saving one parameter, and making it a bit clearer, IMO.